request.js

```diff
@@ -459,6 +464,10 @@ TChannelRequest.prototype.shouldRetryError = function shouldRetryError(err) {
         return false;
     }
 
+    if (self.channel.currentRetryRatio() > self.maxRetryRatio) {
+        return false;
```

Comment:
by default every tchannel will emit metrics for outbound requests, it will emit the amount of attempts and success as well as the amount of retries. 

In theory we can return a custom error class here and then we should have an operational-error stat with the type of error in the name.